### PR TITLE
Closes #335: Notify Slack using proxy

### LINKED ISSUE #335: Add feature to allow slack notification to use proxy
It would be nice to have a feature to notify Slack using a proxy server without modifying the proxy configuration of the whole application.

### PR DESCRIPTION
**Pull request by monicajss**
Solution of issue #335.

### CODE REVIEW

File: promgen/notification/slack.py [hunk 1]
```diff
@@ -37,6 +37,14 @@ class NotificationSlack(NotificationBase):
...
 
     def _send(self, url, data):
+        
+        proxy = self.config('proxies')
+        if proxy is not None and proxy != "":
+            proxies = {
+                'http': proxy,
+                'https': proxy,
+            }
+
```
Comment (kfdm) [inline promgen/notification/slack.py:47]:
```suggestion
        kwargs = {}
        proxy = self.config('proxies')
        if proxy is not None and proxy != "":
            kwargs['proxies'] = {
                'http': proxy,
                'https': proxy,
            }
        
```

File: promgen/notification/slack.py [hunk 2]
```diff
@@ -45,5 +53,11 @@ def _send(self, url, data):
...
-
-        util.post(url, json=json).raise_for_status()
+         
+        if proxy is not None and proxy != "":
+            logger.info('Notifying Slack using proxy')
+            util.post(url, json=json, proxies=proxies).raise_for_status()
+        else:
+            logger.info('Notifying Slack without proxy')
+            util.post(url, json=json).raise_for_status()
+
```
Comment (kfdm) [inline promgen/notification/slack.py:63]:
```suggestion
         
        util.post(url, json=json, **kwargs).raise_for_status()
```

**Review by kfdm — CHANGES_REQUESTED** [review]
Looks good! I think one change I would typically do, is for some of these optional parameters, have an `extra` or `kwargs` (keyword argument) dictionary, to contain the extra parameters such as `proxies`. Then you can call `util.post(url, json=json, **kwargs).raise_for_status()` without needing an additional `if/else` conditional check

File: promgen/notification/slack.py (hunk 2 above)
Comment (monicajss) [reply promgen/notification/slack.py:63]: Thanks for the advice, actually it looks way better using the kwargs, we will update the code, test it and update the pull request, thanks a lot.

**Review by kfdm — APPROVED** [review]
Looks good!

**Pull request merged**